lib/puppet/type/firewalld_custom_service.rb

```diff
@@ -85,7 +85,7 @@
           port_regexp = Regexp.new('^\d+(-\d+)?$')
           raise Puppet::ParseError, "Ports must match #{port_regexp}" unless port_regexp.match?(test_value)
 
-          invalid_ports = test_value.split('-').select { |x| (x == '0') || (x > '65535') }
+          invalid_ports = test_value.split('-').select { |x| (x.to_i < 1) || (x.to_i > 65535) }
```

Comment:
```suggestion
          invalid_ports = test_value.split('-').select { |x| x.to_i.between?(1,65535) }
```

Reply:
You'd need a `!` in there somewhere. I thought of that, but decided it wouldn't actually be more readable.

Reply:
Heh, indeed :-|.

I was just thinking that one `to_i` was better than two. Personally, I find it more readable.